syntax = "proto3";

message OnSceneCreate
{
  uint64 entity = 1;
}

message OnDestroyScene
{
  uint64 entity = 1;
}

message BeforeEnterScene
{
  uint64 entity = 1;
}

message OnEnterScene
{
  uint64 entity = 1;
}

message BeforeLeaveScene
{
  uint64 entity = 1;
}

message OnLeaveScene
{
  uint64 entity = 1;
}

message S2CEnterScene
{
  uint64 entity = 1;
}